airbyte/cloud/workspaces.py

```diff
@@ -450,3 +458,417 @@ def list_destinations(
+
+        Returns:
+            List of CloudCustomSourceDefinition objects
+        """
+        result: list[CloudCustomSourceDefinition] = []
+
+        if custom_connector_type is None or custom_connector_type == "yaml":
+            yaml_definitions = api_util.list_custom_yaml_source_definitions(
+                workspace_id=self.workspace_id,
+                api_root=self.api_root,
+                client_id=self.client_id,
+                client_secret=self.client_secret,
+            )
+            result.extend(
+                CloudCustomSourceDefinition._from_yaml_response(self, d)  # noqa: SLF001
+                for d in yaml_definitions
+                if name is None or d.name == name
+            )
+
+        if custom_connector_type is None or custom_connector_type == "docker":
+            docker_definitions = api_util.list_custom_docker_source_definitions(
+                workspace_id=self.workspace_id,
+                api_root=self.api_root,
+                client_id=self.client_id,
+                client_secret=self.client_secret,
+            )
+            result.extend(
+                CloudCustomSourceDefinition._from_docker_response(self, d)  # noqa: SLF001
+                for d in docker_definitions
+                if name is None or d.name == name
+            )
+
+        return result
+
+    def get_custom_source_definition(
+        self,
+        definition_id: str,
+    ) -> CloudCustomSourceDefinition:
+        """Get a specific custom source definition by ID.
+
+        This method will attempt to fetch as a YAML definition first, then as a Docker
+        definition if not found.
+
+        Args:
+            definition_id: The definition ID
+
+        Returns:
+            CloudCustomSourceDefinition object
+        """
+        try:
+            result = api_util.get_custom_yaml_source_definition(
+                workspace_id=self.workspace_id,
+                definition_id=definition_id,
+                api_root=self.api_root,
+                client_id=self.client_id,
+                client_secret=self.client_secret,
+            )
+            return CloudCustomSourceDefinition._from_yaml_response(self, result)  # noqa: SLF001
+        except Exception:
+            result = api_util.get_custom_docker_source_definition(
+                workspace_id=self.workspace_id,
+                definition_id=definition_id,
+                api_root=self.api_root,
+                client_id=self.client_id,
+                client_secret=self.client_secret,
+            )
+            return CloudCustomSourceDefinition._from_docker_response(self, result)  # noqa: SLF001
+
+    def update_custom_source_definition(
+        self,
+        definition_id: str,
+        *,
+        name: str | None = None,
+        manifest_yaml: dict[str, Any] | Path | str | None = None,
+        docker_tag: str | None = None,
+        pre_validate: bool = True,
+    ) -> CloudCustomSourceDefinition:
```

Comment:
Let's decouple renaming from updating.

1. Add a new "rename_custom_source_definition()" call that just does the rename.
2. Determine the connector type based on whether docker_tag or manifest_yaml is specified. (Exactly one is required.)
3. Based on what type of update is requested we can determine which type to send to our call to 'get_custom_source_definition()'. (See my prior comment about making that a required input arg.)